### DISCUSSION TITLE
Interpolating values defined on vertices of a mesh

### DISCUSSION THREAD

**Discussion by avaruuser** [libigl/libigl#2074] [General]
Hi all,

is there a solid way to interpolate values defined on vertices of a mesh in libigl? For example, I have the values of most of the vertices but some are missing and I want to interpolate the missing values based on the observations that I have.

Thanks for helping out!

**Comment by alecjacobson** [comment]
I'd probably start with k-harmonic solve. So let's say `Eigen::VectorXi b` contains the indices of the vertices where you have known values and `Eigen::VectorXd bc` contain the known values. Then you can solve this with:

```
igl::harmonic(V,F,b,bc,k,W);
```
where `Eigen::VectorXd W` will be a V-long list of values at all vertices.

The positive integer parameter `k` should be set to `1` for a Laplace solution. This has the advantage of being guaranteed (under mild assumptions) to find values bounded by those in `bc`. While `k=2` will trade this property for improved smoothness near interpolated values.
